compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2689,6 +2689,23 @@ pub struct DestructuredConst<'tcx> {
     pub fields: &'tcx [ty::Const<'tcx>],
 }
 
+/// Used in `might_permit_raw_init` to indicate the kind of initialisation
```

Comment:
Just a nit, but I wonder if this is better living in `middle::ty::layout` rather than `middle::ty`? That file is also huge but slightly less large and general than this one.

Reply:
yes